src/libraries/System.Private.CoreLib/src/System/AppContext.AnyOS.cs

```diff
@@ -30,5 +33,51 @@ private static string GetBaseDirectoryCore()
 
             return directory;
         }
+
+        internal static void LogSwitchValues(RuntimeEventSource ev)
+        {
+            if (s_switches is not null)
+            {
+                lock (s_switches)
+                {
+                    foreach (var (k, v) in s_switches)
+                    {
+                        // Convert bool to int because it's cheaper to log (no boxing)
+                        ev.LogAppContextSwitch(k, v ? 1 : 0);
+                    }
+                }
+            }
+
+            if (s_dataStore is not null)
+            {
+                lock (s_dataStore)
+                {
+                    if (s_switches is not null)
+                    {
+                        lock (s_switches)
+                        {
+                            LogDataStore(s_switches);
+                        }
+                    }
+                    else
+                    {
+                        LogDataStore(null);
+                    }
+
+                    void LogDataStore(Dictionary<string, bool>? switches)
```

Comment:
```suggestion
                    static void LogDataStore(Dictionary<string, bool>? switches)
```

Reply:
Captures s_dataStore and `ev` parameter

Reply:
s_dataStore is static; there's no conflict with the method being static and using s_dataStore. But I missed that it captured ev; if it were me, I'd just add `, RuntimeEventSource ev` as an argument and pass it in rather than capturing anything.